core/templates/services/date-time-format.service.spec.ts

```diff
@@ -27,15 +39,17 @@ describe('datetimeformatter', () => {
   beforeEach(() => {
     df = new DateTimeFormatService();
 
-    // Mock Date() to give a time of NOW_MILLIS in GMT. (Unfortunately, there
-    // doesn't seem to be a good way to set the timezone locale directly.)
-    spyOn(window, 'Date').and.callFake(function(millisSinceEpoch = 0) {
+    let MockDateContructor = function(millisSinceEpoch = 0) {
       if (millisSinceEpoch === 0) {
         return new OldDate(NOW_MILLIS);
       } else {
         return new OldDate(millisSinceEpoch);
       }
-    });
+    } as any as MockDateContructorType;
```

Comment:
Why do we need as any here?

Reply:
Because if you look at the type we declared i.e. `MockDateConstructorType`, you'll see we have defined some properties of the function itself. So first we have to do `as any` before assigning that interface to the function.

Reference - https://stackoverflow.com/questions/46501911/typescript-declaring-a-function-with-properties

Reply:
Please add a comment in the code as well.

Reply:
Done!